### ISSUE TITLE
Array formal with queried element size trips internal error

### ISSUE THREAD

**Issue by daviditen** [chapel-lang/chapel#8316]
### Summary of Problem
```
proc f(A: [] real(?s)) { ... }
```

Generates a user-facing error after #14101 was merged, but it would still be better to support this type of queried expression. 

### Steps to Reproduce
```
% chpl test/functions/diten/formalArrayWithQueriedEltSize.chpl
formalArrayWithQueriedEltSize.chpl:1: error: Query expressions are not currently supported in this context
formalArrayWithQueriedEltSize.chpl:1: Function 'f' instantiated as: f(x: [domain(1,int(64),false)] real(64))
```

**Associated Future Test(s):**
[test/functions/diten/formalArrayWithQueriedEltSize.chpl](https://github.com/chapel-lang/chapel/blob/master/test/functions/diten/formalArrayWithQueriedEltSize.chpl)

### Configuration Information
chpl version 1.21.0 pre-release (5747725ba6)

**Comment by bradcray** [comment]
@daviditen: PR #14101 improves upon this situation by turning the internal error into a user-facing error, but additional work is required to actually make it work.  If/when that PR is merged, I propose updating this issue to be more of a feature request if that seems OK with you.

**Comment by daviditen** [comment]
The .future that this issue mentions expected that this pattern should just work, but I updated the description and label to make it more clear that this is a feature request.

**Comment by bradcray** [comment]
I updated the original issue text to describe the updated behavior more accurately now and removed the bug label.

**Comment by dlongnecke-cray** [comment]
Resolved by #23807.

However I have opened #23882 about some remaining work for supporting type queries embedded in array types.
